File: csharp/tests/Training.Module1.Tests/SalesCalculationsTest.cs
using System.Globalization;

namespace Training.Module1.Tests;

public sealed class SalesCalculationsTest
{
    [Fact]
    public void DuplicatedCalculationsCurrentlyProduceTheSameResult()
    {
        decimal unitPrice = 100.00m;

        decimal invoice = SalesCalculations.InvoiceLineTotal(unitPrice, 1, true);
        decimal quote = SalesCalculations.QuoteLineTotal(unitPrice, 1, true);

        // decimal equality ignores scale, so the scale (as in BigDecimal.equals) is checked via text
        Assert.Equal("90.0000", invoice.ToString(CultureInfo.InvariantCulture));
        Assert.Equal(invoice, quote);
    }
}

File: typescript/test/module1/SalesCalculations.test.ts
import { Decimal } from 'decimal.js';
import { describe, expect, it } from 'vitest';
import { SalesCalculations } from '../../src/module1/SalesCalculations.js';

describe('SalesCalculationsTest', () => {
  it('duplicatedCalculationsCurrentlyProduceTheSameResult', () => {
    const unitPrice = new Decimal('100.00');

    const invoice = SalesCalculations.invoiceLineTotal(unitPrice, 1, true);
    const quote = SalesCalculations.quoteLineTotal(unitPrice, 1, true);

    expect(invoice.equals(new Decimal('90.0000'))).toBe(true);
    expect(invoice.toFixed(4)).toBe('90.0000');
    expect(invoice.equals(quote)).toBe(true);
  });
});
